tests/end-to-end/external-solution-integration/internal/scenario/event_mesh/scenario.go

```diff
@@ -1,37 +1,43 @@
 package event_mesh
 
 import (
-	"time"
-
 	"github.com/spf13/pflag"
 
-	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/internal/scenario"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/internal/scenario/event_mesh_evaluate"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/internal/scenario/event_mesh_prepare"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/step"
 )
 
 // E2EScenario executes complete external solution integration test scenario
 type Scenario struct {
+	prepare           event_mesh_prepare.Scenario
+	evaluate          event_mesh_evaluate.Scenario
 	domain            string
 	testID            string
 	skipSSLVerify     bool
 	applicationTenant string
 	applicationGroup  string
-	eventSendDelay    time.Duration
 	logLevel          string
+	testServiceImage  string
 }
 
 // AddFlags adds CLI flags to given FlagSet
 func (s *Scenario) AddFlags(set *pflag.FlagSet) {
 	pflag.StringVar(&s.domain, "domain", "kyma.local", "domain")
-	pflag.StringVar(&s.testID, "testID", "e2e-mesh-ns", "domain")
+	pflag.StringVar(&s.testID, "testID", "external-solution-test", "domain")
 	pflag.BoolVar(&s.skipSSLVerify, "skipSSLVerify", false, "Skip verification of service SSL certificates")
 	pflag.StringVar(&s.applicationTenant, "applicationTenant", "", "Application CR Tenant")
 	pflag.StringVar(&s.applicationGroup, "applicationGroup", "", "Application CR Group")
-	pflag.DurationVar(&s.eventSendDelay, "eventSendDelay", time.Duration(10)*time.Second, "Wait time in seconds before sending the first event, e.g. 5s")
 	pflag.StringVar(&s.logLevel, "logLevel", "info", "Set log level: panic, fatal, error, warn, info, debug, trace")
+	pflag.StringVar(&s.testServiceImage, "testServiceImage", "eu.gcr.io/kyma-project/event-subscriber-tools:PR-8483", "TestServiceImage")
 }
 
 func (s *Scenario) NewState() *state {
-	return &state{
-		E2EState: scenario.E2EState{Domain: s.domain, SkipSSLVerify: s.skipSSLVerify, AppName: s.testID, GatewaySubdomain: "gateway"},
-	}
+	return &state{}
```

Comment:
Why do we need this function `NewState()`?

Reply:
gone